common/kernel/timing.cc

```diff
@@ -1377,115 +878,120 @@ CriticalPath build_critical_path_report(Context *ctx, ClockPair &clocks, const P
-            const auto &b = rb.clock_pair;
-
-            if (a.start.clock.str(ctx) < b.start.clock.str(ctx))
-                return true;
-            if (a.start.clock.str(ctx) > b.start.clock.str(ctx))
-                return false;
-            if (a.start.edge < b.start.edge)
-                return true;
-            if (a.start.edge > b.start.edge)
-                return false;
-            if (a.end.clock.str(ctx) < b.end.clock.str(ctx))
-                return true;
-            if (a.end.clock.str(ctx) > b.end.clock.str(ctx))
-                return false;
-            if (a.end.edge < b.end.edge)
-                return true;
-            return false;
-        });
+    if (clock_reports.empty() && xclock_reports.empty()) {
+        log_info("No Fmax available; no interior timing paths found in design.\n");
+    }
 
-        for (auto &clock : clock_reports) {
-            float target = ctx->setting<float>("target_freq") / 1e6;
-            if (ctx->nets.at(clock.first)->clkconstr)
-                target = 1000 / ctx->getDelayNS(ctx->nets.at(clock.first)->clkconstr->period.minDelay());
-            clock_fmax[clock.first].constraint = target;
-        }
+    std::sort(xclock_reports.begin(), xclock_reports.end(), [&](const CriticalPath &ra, const CriticalPath &rb) {
+        const auto &a = ra.clock_pair;
+        const auto &b = rb.clock_pair;
+
+        if (a.start.clock.str(ctx) < b.start.clock.str(ctx))
+            return true;
+        if (a.start.clock.str(ctx) > b.start.clock.str(ctx))
+            return false;
+        if (a.start.edge < b.start.edge)
+            return true;
+        if (a.start.edge > b.start.edge)
+            return false;
+        if (a.end.clock.str(ctx) < b.end.clock.str(ctx))
+            return true;
+        if (a.end.clock.str(ctx) > b.end.clock.str(ctx))
+            return false;
+        if (a.end.edge < b.end.edge)
+            return true;
+        return false;
+    });
+
+    for (auto &clock : clock_reports) {
+        float target = ctx->setting<float>("target_freq") / 1e6;
+        if (ctx->nets.at(clock.first)->clkconstr)
+            target = 1000 / ctx->getDelayNS(ctx->nets.at(clock.first)->clkconstr->period.minDelay());
+        clock_fmax[clock.first].constraint = target;
     }
 
-    // Print critical paths
+
+    auto print_path = true;
+    auto print_fmax =true;
+
+    auto format_event = [&](const ClockEvent &e, int field_width = 0) {
+        std::string value;
+        if (e.clock == ctx->id("$async$"))
+            value = std::string("<async>");
+        else
+            value = (e.edge == FALLING_EDGE ? std::string("negedge ") : std::string("posedge ")) + e.clock.str(ctx);
+        if (int(value.length()) < field_width)
+            value.insert(value.length(), field_width - int(value.length()), ' ');
+        return value;
+    };
+
+        // Print critical paths
     if (print_path) {
+        if (ctx->idstring_str_to_idx == nullptr) {
```

Comment:
hmm, where is this coming from?

Reply:
It was some debug code. There are more debug prints I will remove once I have validated everything to be equivalent.